gem/ci/build.ps1

```diff
@@ -17,9 +17,10 @@ function Get-ContainerVersion
         git fetch origin 'refs/tags/*:refs/tags/*'
     }
 
-    $gitver = git describe
-    if ($LASTEXITCODE -ne 0) { return '0.0.0' }
-    $gitver -replace '-.*', ''
+    # in a repo with no tags, --always returns a 8 character SHA
+    # when at a specific tag, we get just that number like 2019.2.0.9
+    # otherwise we get something like 2019.1.0.93-6-g4c72515
+    git describe --always
```

Comment:
We still need to do the `-replace` call when it's an actual tag, because this value is used for the package version as well.

But when there's not a tag, just return the `abc123` SHA.

Reply:
The idea/assumption is that when there's no git tag, we're not going to be building any containers that use this value as the package version, because that would fail (presumably we don't ship packages with SHAs as versions).

Reply:
> used for the package version

Do you mean that we use that value to find packages (i.e. PE component urls in Artifactory) when building containers based on packages, rather than from source?

Reply:
Ok - figured it out... this is due to something like https://github.com/puppetlabs/pe-puppetserver/blob/kearney/docker/pe-puppetserver/Dockerfile#L40

Reply:
We'll probably want to do this same pattern in our Makefiles.

Currently for non-component repos we do `VERSION ?= latest` but it might as well be the SHA like we're doing here.